evalml/utils/gen_utils.py

```diff
@@ -520,3 +520,64 @@ def deprecate_arg(old_arg, new_arg, old_value, new_value):
         )
         value_to_use = old_value
     return value_to_use
+
+
+def contains_all_ts_parameters(problem_configuration):
+    """Validates that the problem configuration contains all required keys.
+
+    Args:
+        problem_configuration (dict): Problem configuration.
+
+    Returns:
+        bool, str: True if the configuration contains all parameters. If False, msg is a non-empty
+            string with error message.
+    """
+    required_parameters = {"date_index", "gap", "max_delay", "forecast_horizon"}
+    msg = ""
+    if not problem_configuration or not all(
+        p in problem_configuration for p in required_parameters
+    ):
+        msg = (
+            "problem_configuration must be a dict containing values for at least the date_index, gap, max_delay, "
+            f"and forecast_horizon parameters. Received {problem_configuration}."
+        )
+    return not (msg), msg
+
+
+_validation_result = namedtuple(
+    "TSParameterValidationResult",
+    ("is_valid", "msg", "smallest_split_size", "max_window_size"),
+)
+
+
+def are_ts_parameters_valid_for_split(
+    gap, max_delay, forecast_horizon, n_obs, n_splits
+):
+    """Validates the time series parameters in problem_configuration are compatible with split sizes.
+
+    Args:
+        gap (int): gap value.
+        max_delay (int): max_delay value.
+        forecast_horizon (int): forecast_horizon value.
+        n_obs (int): Number of observations in the dataset.
+        n_splits (int): Number of cross validation splits.
+
+    Returns:
+        TsParameterValidationResult - named tuple with four fields
+            is_valid (bool): True if parameters are valid.
+            msg (str): Contains error message to display. Empty if is_valid.
+            smallest_split_size (int): smallest split size given n_obs and n_splits.
+            max_window_size (int): max window size given gap, max_delay, forecast_horizon.
+    """
+    split_size = n_obs // (n_splits + 1)
+    window_size = gap + max_delay + forecast_horizon
+    msg = ""
+    if split_size <= window_size:
+        msg = (
+            f"Since the data has {n_obs} observations and n_splits={n_splits}, "
+            f"the smallest split would have {split_size} observations. "
+            f"Since {gap + max_delay + forecast_horizon} (gap + max_delay + forecast_horizon)  > {split_size}, "
+            "then at least one of the splits would be empty by the time it reaches the pipeline. "
+            "Please use a smaller number of splits or collect more data."
```

Comment:
I like this error message because it explains why the values passed in will not work! Great work here!

Reply:
Should we change the end of the error message to include this?
"Please use a smaller number of splits, consider reducing one or more of these parameters, or collect more data."

Reply:
I'm on it!